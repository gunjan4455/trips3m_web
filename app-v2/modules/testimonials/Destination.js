import React from 'react';
import PropTpes from 'prop-types';

import { LocationMarkerIcon } from 'helpers/Icon/Icon';

const Destination = ({ breadcrumbs }) => {
  return (
    <div>
      <div className="pl15 pb5 pt5 pr15 flex alignCenter">
        <span className="iblock input-24 verticalMiddle">
          <LocationMarkerIcon />
        </span>
        <div className="iblock pl15 verticalMiddle">
        <p className="iblock f14 fw4 pfc3 pb2">Destinations</p>
        {
          breadcrumbs.length === 4 ?
            <p className="at_destinationDetail fw4 pfc2 f10">
              {`${breadcrumbs[2].name}>${breadcrumbs[3].name}`}
            </p> : null
        }
        </div>
      </div>
    </div>
  );
};

Destination.propTypes = {
  breadcrumbs: PropTpes.array,
};

Destination.defaultProps = {
  breadcrumbs: []
};

export default Destination;

