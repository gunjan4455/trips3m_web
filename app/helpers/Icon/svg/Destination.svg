<?xml version="1.0" encoding="UTF-8"?>
<svg width="160px" height="136px" viewBox="0 0 160 136" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 54.1 (76490) - https://sketchapp.com -->
    <title>illustration/destination</title>
    <desc>Created with Sketch.</desc>
    <g id="illustration/destination" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Group-5">
            <circle id="Oval" fill="#FFF192" cx="80.5" cy="92.5" r="35.5"></circle>
            <g id="Illustration/Destinations">
                <g transform="translate(28.493151, 24.129032)">
                    <path d="M19.9454247,77.884372 C19.9454247,77.884372 23.2484384,72.3276974 25.6637808,75.4596412 C28.0791233,78.5915851 22.7574795,80.8762509 33.8325479,80.8280318 C44.9054247,80.7821089 38.9130959,78.2127209 48.9865205,82.0794317 C59.0599452,85.9484386 44.752,84.7245925 46.9306301,88.2170314 C49.1092603,91.7094703 48.681863,106.80893 42.601863,98.5772446 C36.521863,90.3455592 39.6341918,98.8298207 31.4654247,91.3788252 C23.292274,83.9301258 17.5037808,91.7944277 19.9454247,77.884372 Z" id="Stroke-1" stroke="#000000" fill="#009688" stroke-linecap="round" stroke-linejoin="round"></path>
                    <polygon id="Stroke-5" stroke="#000000" stroke-linecap="round" stroke-linejoin="round" points="25.4226849 110.949801 1.09391781 106.984355 12.7059726 65.3483304 31.4566575 67.6651422"></polygon>
                    <polygon id="Stroke-7" stroke="#000000" stroke-linecap="round" stroke-linejoin="round" points="50.8530411 106.985274 25.4218082 110.950719 31.4557808 67.6660607 50.8530411 65.3469527"></polygon>
                    <path d="M72.4475616,83.3533338 C68.4366027,78.921771 63.283726,72.8622403 58.4398904,66.2722997" id="Stroke-9" stroke="#000000" stroke-linecap="round" stroke-linejoin="round"></path>
                    <path d="M76.2831781,110.954163 L50.8563288,106.98183 L50.8563288,65.352693 L58.4398904,66.2711517 C63.283726,72.8610922 68.4366027,78.9229191 72.4475616,83.3521857 L74.573589,98.6927407 L75.2727671,103.69834 L76.2831781,110.954163 Z" id="Stroke-11" stroke="#000000" stroke-linecap="round" stroke-linejoin="round"></path>
                    <path d="M92.1286575,76.5567401 C86.8245479,82.8941045 82.134137,87.8308195 80.2492055,89.7825441 C79.8108493,89.9914934 79.4382466,90.2876963 79.1533151,90.517311 C78.4760548,89.828467 75.8875616,87.2131561 72.4486575,83.3533338" id="Stroke-13" stroke="#000000" stroke-linecap="round" stroke-linejoin="round"></path>
                    <path d="M100.611068,106.980682 L76.2823014,110.953015 L75.2718904,103.697192 L74.5727123,98.6938888 L72.4466849,83.3533338 C75.8877808,87.2131561 78.4740822,89.828467 79.1535342,90.517311 C79.4384658,90.2876963 79.8110685,89.9914934 80.2494247,89.7825441 C82.1343562,87.8308195 86.8247671,82.8941045 92.1288767,76.5567401 L100.611068,106.980682 Z" id="Stroke-15" stroke="#000000" stroke-linecap="round" stroke-linejoin="round"></path>
                    <path d="M79.3290959,52.5796891 C69.5844384,52.5796891 61.6830685,44.304377 61.6830685,34.0934134 C61.6830685,23.8847459 69.5844384,15.6071376 79.3290959,15.6071376 C89.0737534,15.6071376 96.9751233,23.8847459 96.9751233,34.0934134 C96.9751233,44.304377 89.0737534,52.5796891 79.3290959,52.5796891 M79.3290959,1.14830288 C61.9592329,1.14830288 47.8792329,15.8964521 47.8792329,34.0934134 C47.8792329,52.2880785 79.3290959,83.6258864 79.3290959,83.6258864 C79.3290959,83.6258864 110.776767,52.2880785 110.776767,34.0934134 C110.776767,15.8964521 96.6967671,1.14830288 79.3290959,1.14830288" id="Fill-17" fill="#009688"></path>
                    <path d="M79.3290959,52.5796891 C69.5844384,52.5796891 61.6830685,44.304377 61.6830685,34.0934134 C61.6830685,23.8847459 69.5844384,15.6071376 79.3290959,15.6071376 C89.0737534,15.6071376 96.9751233,23.8847459 96.9751233,34.0934134 C96.9751233,44.304377 89.0737534,52.5796891 79.3290959,52.5796891 Z M79.3290959,1.14830288 C61.9592329,1.14830288 47.8792329,15.8964521 47.8792329,34.0934134 C47.8792329,52.2880785 79.3290959,83.6258864 79.3290959,83.6258864 C79.3290959,83.6258864 110.776767,52.2880785 110.776767,34.0934134 C110.776767,15.8964521 96.6967671,1.14830288 79.3290959,1.14830288 Z" id="Stroke-19" stroke="#000000" stroke-linecap="round" stroke-linejoin="round"></path>
                </g>
            </g>
        </g>
    </g>
</svg>