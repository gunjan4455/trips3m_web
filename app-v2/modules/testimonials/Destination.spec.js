import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Destination from './Destination';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  breadcrumbs: [
    { name: 'Home', url: '/' },
    { name: 'Testimonials', url: '/testimonials' },
  ]
};

const shallowWrapperDestination = props => {
  return shallow(<Destination {...props} />);
};

describe('Testimonials module', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperDestination(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render destinationDetail', () => {
    const wrapper = shallowWrapperDestination(props);
    expect(wrapper.find('.at_destinationDetail').exists()).toBeFalsy();
  });

  it('should render proper destinationDetail ', () => {
    props.breadcrumbs.push(
      { name: 'India', url: '' },
      { name: 'kerala', url: '/testimonials/kerala' },
    );
    const wrapper = shallowWrapperDestination(props);
    expect(wrapper.find('.at_destinationDetail').text()).toBe('India>kerala');
  });
});
