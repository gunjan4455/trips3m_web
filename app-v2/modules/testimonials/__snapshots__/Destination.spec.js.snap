// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Destination
    breadcrumbs={
      Array [
        Object {
          "name": "Home",
          "url": "/",
        },
        Object {
          "name": "Testimonials",
          "url": "/testimonials",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pl15 pb5 pt5 pr15 flex alignCenter"
      >
        <span
          className="iblock input-24 verticalMiddle"
        >
          <IconSvg />
        </span>
        <div
          className="iblock pl15 verticalMiddle"
        >
          <p
            className="iblock f14 fw4 pfc3 pb2"
          >
            Destinations
          </p>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="iblock input-24 verticalMiddle"
          >
            <IconSvg />
          </span>,
          <div
            className="iblock pl15 verticalMiddle"
          >
            <p
              className="iblock f14 fw4 pfc3 pb2"
            >
              Destinations
            </p>
          </div>,
        ],
        "className": "pl15 pb5 pt5 pr15 flex alignCenter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "iblock input-24 verticalMiddle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="iblock f14 fw4 pfc3 pb2"
              >
                Destinations
              </p>,
              null,
            ],
            "className": "iblock pl15 verticalMiddle",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Destinations",
                "className": "iblock f14 fw4 pfc3 pb2",
              },
              "ref": null,
              "rendered": "Destinations",
              "type": "p",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pl15 pb5 pt5 pr15 flex alignCenter"
        >
          <span
            className="iblock input-24 verticalMiddle"
          >
            <IconSvg />
          </span>
          <div
            className="iblock pl15 verticalMiddle"
          >
            <p
              className="iblock f14 fw4 pfc3 pb2"
            >
              Destinations
            </p>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="iblock input-24 verticalMiddle"
            >
              <IconSvg />
            </span>,
            <div
              className="iblock pl15 verticalMiddle"
            >
              <p
                className="iblock f14 fw4 pfc3 pb2"
              >
                Destinations
              </p>
            </div>,
          ],
          "className": "pl15 pb5 pt5 pr15 flex alignCenter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <IconSvg />,
              "className": "iblock input-24 verticalMiddle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="iblock f14 fw4 pfc3 pb2"
                >
                  Destinations
                </p>,
                null,
              ],
              "className": "iblock pl15 verticalMiddle",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Destinations",
                  "className": "iblock f14 fw4 pfc3 pb2",
                },
                "ref": null,
                "rendered": "Destinations",
                "type": "p",
              },
              null,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
